packages/ui/react-ui-table/src/components/Cell/Cell.tsx

```diff
@@ -32,21 +32,22 @@ const FocusableCell = <TData extends RowData, TValue>({ cell, children }: CellPr
 
     const cellIndex = Array.from(currentRow.cells || []).indexOf(currentTarget);
 
-    const focusNextCell = (nextRow: HTMLTableRowElement) => {
-      const nextCell = nextRow.cells[cellIndex];
-      nextCell.focus();
-    };
-
     const nextSibling = currentRow.nextElementSibling;
-    if (nextSibling instanceof HTMLTableRowElement) {
-      focusNextCell(nextSibling);
-    } else {
+    if (!(nextSibling instanceof HTMLTableRowElement)) {
       let depth = 0;
       // Poll for the new 'add row' row to be added to the DOM.
       const pollForNewRow = () => {
         const newNextSibling = currentRow.nextElementSibling;
         if (newNextSibling instanceof HTMLTableRowElement) {
-          focusNextCell(newNextSibling);
+          const nextCell = newNextSibling.cells[cellIndex];
+          // Find the input within next cell and focus it.
+          const input = nextCell.querySelector('input');
+          if (input) {
+            input.focus();
+          }
+
+          // Scroll current row into view after it's been hidden behind the sticky add-row row.
+          currentRow.scrollIntoView({ behavior: 'smooth', block: 'center' });
```

Comment:
see my sheets example. this doesn't quite work perfectly since it doesn't include the border. i have a util for this. pls take a look later and add a TODO.

scrollIntoView.ts